lib/Cake/View/Helper/FormHelper.php

```diff
@@ -1638,7 +1645,12 @@ public function radio($fieldName, $options = array(), $attributes = array()) {
 			$between = '';
 		}
 		if ($legend) {
-			$out = $this->Html->useTag('fieldset', '', $this->Html->useTag('legend', $legend) . $between . $out);
+			if (is_string($fieldset)) {
+				$fieldsetClass = sprintf(' class="%s"', $fieldset);
```

Comment:
Ok, I see :) Just to be one the same page with you:

> Could we use an array of attributes instead of string smashing

Should I use _extractOption() for reading the attribute value instead?

> You are not correctly HTML escaping the class name here.

Should I use `$fieldsetClass = sprintf(' class="%s"', h($fieldset));` instead?